blog/2015-09-30-HowLuckyAreYourRandomSeeds.adoc

```diff
@@ -0,0 +1,141 @@
+infeasible runs (4 of them on the `comp05` dataset and 1 on `comp02`) as compared to Tabu Search's 13 infeasible runs.
+** There were no outliers on the hard score level.
+** There were 2 outliers in runs of the Late Acceptance solver configuration:
+the relative percentage change in `comp07` being 25%, in `comp01` 104%.
+** The largest relative percentage score change between runs was about 37.5% (Tabu Search on `comp07`)
+
+* https://docs.jboss.org/optaplanner/blog/benchmark/2015-09-30/machinereassignment[Machine Reassignment benchmark report]
+** This benchmark solves 10 datasets of the same use-case with 2 different solver configurations.
+** There were no differences in hard scores between runs.
+** The only outlier was in a run of the Late Acceptance solver configuration in `model_b_7`:
+the relative percentage change was still only ~3%
+** The largest relative percentage score change between runs was about 8% (Tabu Search on `model_b_3`)
+** All the problem benchmark score distribution charts, the reliability comparison of different solver configurations
+is visualized very nicely (see for example `model_b_8` -- Late Acceptance is a lot more reliable in that dataset).
+** In this use-case, we can see a significant triumph of one solver configuration against the other:
+f.e. on the datasets `model_b_5` and `model_b_9`
+(the worst Tabu Search run on these datasets wins over the best Late Acceptance run by a big margin).
+
+=== Summary
+
+The relative percentage score change can seem large
+(the largest one in the General benchmark was 13%, in the CurriculumCourse 104%),
+but this measurement varies wildly per dataset
+-- there are datasets, where all solver configuration runs vary greatly
+(`comp01` of CurriculumCourse) and then there are datasets,
+where all solver configuration runs vary only by a tiny bit (`model_b_4` of MachineReassignment).
+This indicates that the variance is more influenced by the dataset than by the random seed or solver configuration.
+
+The various outliers suggest that there are "lucky" and "unlucky" random seeds,
+but they don't seem to influence the results in any significant way *in general*.
+Again, there are specific datasets and use-cases which are more or less prone to changes in random seeds
+-- we can can make use of statistical benchmarking to confirm if this is or isn't our case.
+
+== Do we need a lucky random seed to get optimal results?
+
+We need more than just a lucky random seed to get optimal results.
+My personal opinion is, that our test results were *inconclusive*.
+There are several indicators:
+
+* We can't tell which random seed is better, because even a bad performing one in a 5 minute run could generate awesome moves in a 10 minute run.
+* On some datasets, our scores were stable whereas on others, the results varied wildly.
+
+== Conclusion
+
+In production, definitely use the `PRODUCTION` environment mode and let OptaPlanner choose a random seed on random.
```

Comment:
Depends on their preferences, I think as far as the random seed issue goes, the `PRODUCTION` environment produces better results on average, but again, that's just my (subjective) opinion.